### ISSUE TITLE
ADD A NEW SWITCH TO POSTPROCESSOR WHEN=before_merge

### ISSUE THREAD

**Issue by annonymouscoder71** [yt-dlp/yt-dlp#8266]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.09.24** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

This feature has been asked for multiple times in the past. So when using +, yt-dlp invoked the merger directly and when using comma it does not invoke the merger I want to do it like this. I want to invoke the merger but not immediately after video download. I want to pass them through other processes before merging. Also I want to know how do I get the correct filename when using + operator as it appends the format id at the end.

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by annonymouscoder71** [comment]
Alternatively if there is any way to use merger with the comma operator afterwards. (Instead of when=before_merge)

**Comment by bashonly** [comment]
Hmm #7905 why would you need to run a postprocessor before merge

**Comment by annonymouscoder71** [comment]
Trimming audio/video or subtitles correction/translation.

**Comment by bashonly** [comment]
Why do those need to be done before merge?

**Comment by annonymouscoder71** [comment]
It's just more convenient with the inbuilt merger to merge video/audio together but the difference in naming scheme when using + and , makes it difficult to use in a batch script.

**Comment by bashonly** [comment]
> subtitles correction/translation.

This seems entirely unrelated to what you are asking. Subtitles are downloaded separately and have their own filepath that is not impacted by `merged+formats`. You can just do this
`--use-postprocessor "YourSubtitleCorrector:when=before_dl"`

> Trimming audio/video 
> It's just more convenient

Why would it be more convenient to trim the video after download, but before merge or any other postprocessing? You are having to manage more files and making everything more complicated this way.

Come up with a valid use case or else I have to assume this is a duplicate of #7905

**Comment by annonymouscoder71** [comment]
Thank you, i didn't know that subtitles could be pre-processed.
